api/src/transactions.rs

```diff
@@ -1191,7 +1190,20 @@ impl TransactionsApi {
 
         // Simulate transaction
         let state_view = self.context.latest_state_view_poem(&ledger_info)?;
-        let (vm_status, output) = AptosVM::simulate_signed_transaction(&txn, &state_view);
+
+        let vm = AptosVM::new(&state_view.as_move_resolver()).for_simulation();
+
+        let output = vm
+            .simulate_signed_transaction(&txn, &state_view)
+            .map_err(|vm_status| {
+                SubmitTransactionError::internal_with_code(
+                    format!("Failed to simulate a transaction: {}", vm_status),
+                    AptosErrorCode::InternalError,
+                    &ledger_info,
+                )
+            })?;
```

Comment:
I was trying to reason about how the early return changes the semantics of this method (e.g. there is exe_status below but it goes according to output.status, there is the other place you removed where vm_status is used, but also the fact that the rest of the code won't even be performed), can you help me out?

Reply:
So here we don't return a JSON with transaction information, such as events, gas, etc. I have basically replaced `Result<(VMStatus, VMOutput), VMStatus>` with `Result<VMOutput, VMStatus>` which might not be desired actually... I think I will actually return the old flow, good point about that, thanks!

Reply:
Actually now it is `(VMStatus, VMOutput)`, because made the verification of signature an assert and aggregator V1 deltas to never fail.